apps/shell-ui/src/connection/errors.ts

```
@@ -0,0 +1,63 @@
+ * Bound an operation and cancel its client-side state before reporting timeout.
+ * Late completion is deliberately ignored after timeout so it cannot revive a
+ * login that the caller has already treated as failed.
+ */
+export function withTimeout<T>(
+	operation: Promise<T>,
+	timeoutMs: number,
+	timeoutError: Error,
+	cleanup: () => Promise<unknown>,
+): Promise<T> {
+	return new Promise<T>((resolve, reject) => {
+		let timeout: ReturnType<typeof setTimeout> | undefined;
```

ESLint

[error] 'timeout' is never reassigned. Use 'const' instead.

(prefer-const)